engines/bastion_katello/app/assets/javascripts/bastion_katello/errata/details/erratum-repositories.controller.js

```diff
@@ -46,10 +49,20 @@ function ($scope, $q, Nutupane, Repository, Environment, ContentView, CurrentOrg
 
     $scope.filterErrata = function () {
         params['environment_id'] = $scope.environmentFilter;
-        params['content_view_version_id'] = $scope.contentViewFilter;
 
         if ($scope.contentViewFilter) {
-            params['content_view_version_id'] = _.map($scope.contentViewFilter.versions, 'id');
+            version = _.find($scope.contentViewFilter.versions, function(version) {
+                // Find the version belonging to the environment specified by the enviroment filter
+                env = _.find(version.environment_ids, function(env_id) {
+                    return env_id === $scope.environmentFilter;
+                });
+
+                return !angular.isUndefined(env);
+            });
+
+            if (!angular.isUndefined(version)) {
+                params['content_view_version_id'] = version.id;
+            }
```

Comment:
@cfouant , @parthaa - 
Question:
Could we `delete params['content_view_version_id']` if the cv version is undefined? Changing environment filter (to an env without this cv) leads to no results, mostly cause the API call is made with the old cv filter and new env filter.